src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/VsActiveEditorContextTracker.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.VisualStudio.ProjectSystem.LanguageServices;
+using Microsoft.VisualStudio.TextManager.Interop;
+using HierarchyId = Microsoft.VisualStudio.Shell.HierarchyId;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.LanguageServices
+{
+    /// <summary>
+    ///     Tracks the "active" context for the editor by handling the VSHPROPID_ActiveIntellisenseProjectContext hierarchy property.
+    /// </summary>
+    [Export(typeof(IActiveIntellisenseProjectProvider))]
+    [ExportProjectNodeComService(typeof(IVsContainedLanguageProjectNameProvider))]
+    [AppliesTo(ProjectCapability.DotNetLanguageService)]
+    internal class VsActiveEditorContextTracker : IActiveIntellisenseProjectProvider, IVsContainedLanguageProjectNameProvider
+    {
+        private readonly IActiveEditorContextTracker _activeEditorContextTracker;
+
+        // UnconfiguredProject is only included for scoping reasons.
+        [ImportingConstructor]
+        public VsActiveEditorContextTracker(UnconfiguredProject? project, IActiveEditorContextTracker activeEditorContextTracker)
+        {
+            _activeEditorContextTracker = activeEditorContextTracker;
+        }
+
+        public string? ActiveIntellisenseProjectContext
+        {
+            get { return _activeEditorContextTracker.ActiveIntellisenseProjectContext; }
+            set { _activeEditorContextTracker.ActiveIntellisenseProjectContext = value; }
+        }
+
+        public int GetProjectName(uint itemid, out string? pbstrProjectName)
+        {
+            if (itemid == HierarchyId.Nil || itemid == HierarchyId.Selection || itemid == HierarchyId.Empty)
+            {
+                pbstrProjectName = null;
+                return HResult.InvalidArg;
+            }
+
+            pbstrProjectName = ActiveIntellisenseProjectContext;
+
+            return HResult.OK;
+        }
+
+        // Pass-through for unit testing.
+        internal bool IsActiveEditorContext(string contextId) => _activeEditorContextTracker.IsActiveEditorContext(contextId);
+
+        // Pass-through for unit testing.
+        internal IDisposable RegisterContext(string contextId) => _activeEditorContextTracker.RegisterContext(contextId);
```

Comment:
These are internal in an internal class because they never are intended to be public (only used for unit testing).

Reply:
I'm a bit confused by what a mock would be providing here. Right now, the unit tests create actual instances of both `VsActiveEditorContextTracker` and `ActiveEditorContextTracker`. Both instances are accessible to the tests. I did it this way to reduce the changes necessary to the tests. Instead, I could update the tests to access both class instances directly. My original changes here were actually to expose the composed `ActiveEditorContextTracker` within the class as internal, but that requires adding `._activeEditorContextTracker` to every call of these methods in the tests.